src/BuildScriptGenerator/PlatformInstallerBase.cs

```diff
@@ -109,8 +109,19 @@ public virtual void InstallPlatformSpecificSkeletonDependencies(StringBuilder st
                 .AppendLine($"cd {versionDirInTemp}")
                 .AppendLine("PLATFORM_BINARY_DOWNLOAD_START=$SECONDS")
                 .AppendLine($"platformName=\"{platformName}\"")
-                .AppendLine("echo \"Detecting image debian flavor: $DEBIAN_FLAVOR.\"")
-                .AppendLine($"if [ \"$DEBIAN_FLAVOR\" == \"{OsTypes.DebianStretch}\" ]; then")
+                .AppendLine($"if [ -z \"$DEBIAN_FLAVOR\" ]; then")
```

Comment:
Just brainstorming here: is there any reason we can't just set `DEBIAN_FLAVOR` equal to the value of `this.CommonOptions.DebianFlavor` here? At this point in the code, `this.CommonOptions.DebianFlavor` will either be set to the `DEBIAN_FLAVOR` environment variable, the value of the `.ostype` file, or will have thrown an exception that terminated the process if neither were set/found. Is it possible for the user to have changed either of these conditions between the time we checked `this.CommonOptions.DebianFlavor` in the `BuildCommand` class and when this build script is ran? Both code snippets are executed _after_ the user runs `oryx build`, so I don't see a way that the behavior could change, but would love to hear what you think because I might be missing something.

Reply:
Yeah this is definitely correct! It should not be possible to get to this stage and not have the CommonOptions.DebianFlavor populated. It's a good point to just simplify this logic and set it in this script from the get-go.